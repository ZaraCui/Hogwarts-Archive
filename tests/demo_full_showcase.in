ADD STUDENT Harry Potter
ADD STUDENT Hermione Granger
ADD STUDENT Ron Weasley
ADD STUDENT Draco Malfoy

ADD COLLECTION tests/data/books_small.csv
ADD COLLECTION tests/data/books_dupes.csv
ADD SPELLBOOK tests/data/books_partial.csv 2001

LIST ALL
LIST ALL LONG
LIST TYPES
LIST INVENTORS
TYPE Charm
INVENTOR Minerva McGonagall

RENT 100000 1001
RENT 100001 1001
SPELLBOOK 1001 LONG
LIST AVAILABLE
RELINQUISH 100001 1001
RELINQUISH 100000 1001
SPELLBOOK HISTORY 1001

STUDENT 100000
STUDENT SPELLBOOKS 100000
STUDENT HISTORY 100000

RENT 100002 1002
RENT 100001 1002
RELINQUISH ALL 100002

COMMON 100000 100001 100002
SAVE COLLECTION tests/data/save_demo.csv

EXIT

