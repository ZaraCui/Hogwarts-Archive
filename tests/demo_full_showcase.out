user: Success.

user: Success.

user: Success.

user: Success.

user: 12 spellbooks successfully added.

user: 5 spellbooks successfully added.

user: Spellbook already exists in system.

user: Levitation Charms (Helga Hufflepuff)
Dark Arts Basics (Salazar Slytherin)
Patronus Guide (Godric Gryffindor)
Transfiguration I (Minerva McGonagall)
Advanced Charms (Filius Flitwick)
Herbology Field Notes (Newt Scamander)
Polyjuice Manual (Libatius Borage)
Transfiguration i (Minerva McGonagall)
patronus guide (Godric Gryffindor)
Dark Arts Basics (Salazar Slytherin)
Levitation Charms (Helga Hufflepuff)
Advanced Charms (Filius Flitwick)
Advanced Charms (Filius Flitwick)
Dark Arts Basics (Salazar Slytherin)
Transfiguration II (Minerva McGonagall)
Herbology Advanced (Newt Scamander)
Polyjuice Advanced (Libatius Borage)

user: 1001: Levitation Charms (Helga Hufflepuff, Charm)
Currently available.

1002: Dark Arts Basics (Salazar Slytherin, Curse)
Currently available.

1003: Patronus Guide (Godric Gryffindor, Defense)
Currently available.

1004: Transfiguration I (Minerva McGonagall, Transfiguration)
Currently available.

1005: Advanced Charms (Filius Flitwick, Charm)
Currently available.

1006: Herbology Field Notes (Newt Scamander, Herbology)
Currently available.

1007: Polyjuice Manual (Libatius Borage, Potion)
Currently available.

1008: Transfiguration i (Minerva McGonagall, TRANSFIGURATION)
Currently available.

1009: patronus guide (Godric Gryffindor, defense)
Currently available.

1010: Dark Arts Basics (Salazar Slytherin, Curse)
Currently available.

1011: Levitation Charms (Helga Hufflepuff, CHARM)
Currently available.

1012: Advanced Charms (Filius Flitwick, Charm)
Currently available.

2001: Advanced Charms (Filius Flitwick, Charm)
Currently available.

2002: Dark Arts Basics (Salazar Slytherin, Curse)
Currently available.

2003: Transfiguration II (Minerva McGonagall, Transfiguration)
Currently available.

2004: Herbology Advanced (Newt Scamander, Herbology)
Currently available.

2005: Polyjuice Advanced (Libatius Borage, Potion)
Currently available.

user: Charm
Curse
Defense
Herbology
Potion
Transfiguration

user: Filius Flitwick
Godric Gryffindor
Helga Hufflepuff
Libatius Borage
Minerva McGonagall
Newt Scamander
Salazar Slytherin

user: Advanced Charms (Filius Flitwick)
Advanced Charms (Filius Flitwick)
Advanced Charms (Filius Flitwick)
Levitation Charms (Helga Hufflepuff)
Levitation Charms (Helga Hufflepuff)

user: Transfiguration I (Minerva McGonagall)
Transfiguration i (Minerva McGonagall)
Transfiguration II (Minerva McGonagall)

user: Success.

user: Spellbook is currently unavailable.

user: 1001: Levitation Charms (Helga Hufflepuff, Charm)
Rented by: 100000.

user: Dark Arts Basics (Salazar Slytherin)
Patronus Guide (Godric Gryffindor)
Transfiguration I (Minerva McGonagall)
Advanced Charms (Filius Flitwick)
Herbology Field Notes (Newt Scamander)
Polyjuice Manual (Libatius Borage)
Transfiguration i (Minerva McGonagall)
patronus guide (Godric Gryffindor)
Dark Arts Basics (Salazar Slytherin)
Levitation Charms (Helga Hufflepuff)
Advanced Charms (Filius Flitwick)
Advanced Charms (Filius Flitwick)
Dark Arts Basics (Salazar Slytherin)
Transfiguration II (Minerva McGonagall)
Herbology Advanced (Newt Scamander)
Polyjuice Advanced (Libatius Borage)

user: Unable to return spellbook.

user: Success.

user: 100000

user: 100000: Harry Potter

user: Student not currently renting.

user: Levitation Charms (Helga Hufflepuff)

user: Success.

user: Spellbook is currently unavailable.

user: Success.

user: No common spellbooks.

user: Success.

user: Ending Archive process.
